src/antsibull_core/logging.py

```diff
@@ -243,4 +253,156 @@ def initialize_app_logging() -> None:
     twiggy.quick_setup(min_level=_level)
 
 
-__all__ = ("log", "initialize_app_logging")
+class Logger(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def debug(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def info(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def notice(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def warning(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def error(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def critical(self, format_spec: str, *args, **kwargs) -> None:
+        pass
+
+    @abc.abstractmethod
+    def fields(self, **kwargs) -> Logger:
+        pass
+
+
+class TwiggyLogger(Logger):
+    def __init__(self, logger: twiggy.logger.Logger) -> None:
+        self.logger = logger
+
+    def debug(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.debug(format_spec, *args, **kwargs)
+
+    def info(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.info(format_spec, *args, **kwargs)
+
+    def notice(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.notice(format_spec, *args, **kwargs)
+
+    def warning(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.warning(format_spec, *args, **kwargs)
+
+    def error(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.error(format_spec, *args, **kwargs)
+
+    def critical(self, format_spec: str, *args, **kwargs) -> None:
+        self.logger.critical(format_spec, *args, **kwargs)
+
+    def fields(self, **kwargs) -> Logger:
+        return TwiggyLogger(self.logger.fields(**kwargs))
+
+
+@dataclass
+class LogMessage:
```

Comment:
Maybe underscore-prefix this class name to make it clearer that it's not for public use.